plugin/src/main/cpp/register_types.cpp

```diff
@@ -86,109 +86,165 @@
 
 using namespace godot;
 
+static Vector<OpenXRExtensionWrapperExtension *> extensions_registered;
+static Vector<OpenXRExtensionWrapperExtension *> extensions_singletons;
+
+static void _add_bool_project_setting(ProjectSettings *p_project_settings, const String &p_name, bool p_default_value) {
+	if (!p_project_settings->has_setting(p_name)) {
+		p_project_settings->set_setting(p_name, p_default_value);
+	}
+
+	p_project_settings->set_initial_value(p_name, p_default_value);
+	p_project_settings->set_as_basic(p_name, false);
+	Dictionary property_info;
+	property_info["name"] = p_name;
+	property_info["type"] = Variant::Type::BOOL;
+	property_info["hint"] = PROPERTY_HINT_NONE;
+	p_project_settings->add_property_info(property_info);
+}
+
+static inline bool _get_bool_project_setting(const String &p_name) {
+	return (bool)ProjectSettings::get_singleton()->get_setting_with_override(p_name);
+}
+
+static void _register_extension_with_openxr(OpenXRExtensionWrapperExtension *p_extension) {
+	p_extension->register_extension_wrapper();
+	extensions_registered.push_back(p_extension);
+}
+
+static void _register_extension_as_singleton(const StringName &p_name, OpenXRExtensionWrapperExtension *p_extension) {
+	Engine::get_singleton()->register_singleton(p_name, p_extension);
```

Comment:
Could you use `get_class_static()` (or related `GDCLASS` functions) to get the class name of the extension?